src/Compilers/CSharp/Test/Semantic/Semantics/ConstantTests.cs

```diff
@@ -2904,7 +2907,10 @@ public static void Bug1098197_02()
     Diagnostic(ErrorCode.ERR_InvalidExprTerm, ")").WithArguments(")").WithLocation(2, 16),
     // (2,18): error CS1023: Embedded statement cannot be a declaration or labeled statement
     // void f() { if () const int i = 0; }
-    Diagnostic(ErrorCode.ERR_BadEmbeddedStmt, "const int i = 0;").WithLocation(2, 18)
+    Diagnostic(ErrorCode.ERR_BadEmbeddedStmt, "const int i = 0;").WithLocation(2, 18),
+    // (2,28): warning CS0219: The variable 'i' is assigned but its value is never used
+    // void f() { if () const int i = 0; }
+    Diagnostic(ErrorCode.WRN_UnreferencedVarAssg, "i").WithArguments("i").WithLocation(2, 28)
                 );
```

Comment:
The warning is reported for declarations within embedded statements. It is an error to use local declaration statement as an embedded statement, see the error above the warning. Therefore, the warning is not a breaking change.